### PR TITLE
Issue/23/histogram

### PR DESCRIPTION
**Pull request by aimalz**
I implemented the histogram parametrization of the PDF object and updated the demo notebook to show its performance.

### CODE REVIEW

**Comment by drphilmarshall** [comment]
Great! I don't have time to review this now, but am looking fwd to it. In
the meantime, can you please review the sphinx doc PR so we can merge it
and move on with the notebooks in the right place? Thanks!

File: qp/pdf.py [hunk 1]
```diff
@@ -124,31 +186,42 @@ def interpolate(self, vb=True):
...
 
-        self.difs = self.quantiles[1:]-self.quantiles[:-1]
-        self.mids = (self.quantiles[1:]+self.quantiles[:-1])/2.
-        self.quantvals = (1.0/(len(self.quantiles)+1))/self.difs
+        if using == 'histogram':
+            # First find the histogram if none exists:
+            if self.histogram is None:
+                self.histogram = self.histogramize()
+
+            self.mids = (self.histogram[0][1:]+self.histogram[0][:-1])/2.
```
Comment (drphilmarshall) [inline qp/pdf.py:234]: Here we over-write the self.mids and self.vals, and indeed the interpolator. I think that's OK, but we might want to have the PDF track what its current approximation is. `self.approximation = "histogram"` etc. A risk with the current implementation: is it possible to end up using the wrong interpolator?

**Review by drphilmarshall — COMMENTED** [review]
Good stuff - at first I wondered whether we should be using the numpy or matplotlib histogram functions, but making and plotting by hand open up the possibility of using irregular bins later.

File: qp/pdf.py (hunk 1 above)
Comment (aimalz) [reply qp/pdf.py:234]: I introduced `PDF.last` to store what parametrization was calculated most recently.  Now the interpolator uses that unless something else is specified, and it prints which parametrization that is.

**Comment by drphilmarshall** [comment]
Sounds good!

File: qp/pdf.py (hunk 1 above)
Comment (drphilmarshall) [reply qp/pdf.py:234]: Sounds good - thanks!

**Comment by aimalz** [comment]
Oh no, there are conflicts!  I don't recall, do these become your problem as the reviewer or do I need to resolve them somehow before the merge can be completed?

**Comment by drphilmarshall** [comment]
I'm on the case...

**Review by drphilmarshall — APPROVED** [review]

**Comment by aimalz** [comment]
This merge didn't pass the continuous integration test because of a spurious newline in pdf.py, which I fixed (not realizing where it came from at the time) in #37.

**Pull request merged**